### PR TITLE
Cs2 patch

### PR DESCRIPTION
**Pull request by hassount**
Port GM bug fix / nsq-sq weight and unpin threads with -lp x

### CODE REVIEW

File: Source/Lib/Encoder/Codec/EbProductCodingLoop.c [hunk 1]
```diff
@@ -32,6 +32,19 @@
...
+                                         -6,  -5,  -4,  -3,  -2,  -1,   0,   0,   0,   0,   0,   0,   0,
+                                          0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,
+                                          0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0    };
+// nsq_weight
+static int32_t nsq_weight_per_qp[64] = { -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,
+                                         -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -5,  -4,  -4,
+                                         -3,  -3,  -2,  -2,  -1,  -1,   0,   0,   0,   0,   0,   0,   0,
+                                          0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,
+                                          0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0    };
+
```
Comment (mafonsoN) [inline Source/Lib/Encoder/Codec/EbProductCodingLoop.c]: How were these determined?

File: Source/Lib/Encoder/Codec/EbProductCodingLoop.c [hunk 2]
```diff
@@ -7909,7 +7922,7 @@ uint8_t update_skip_nsq_shapes(SequenceControlSet *scs_ptr, PictureControlSet *p
     if (sq_weight == (uint32_t)~0) return skip_nsq;
 
     // use an aggressive threshold for QP 20
```
Comment (mafonsoN) [inline Source/Lib/Encoder/Codec/EbProductCodingLoop.c]: Change the comment

File: Source/Lib/Encoder/Globals/EbEncHandle.c [hunk 3]
```diff
@@ -1644,9 +1647,9 @@ EB_API EbErrorType eb_init_encoder(EbComponentType *svt_enc_component)
     /************************************
     * Thread Handles
     ************************************/
-    EbSvtAv1EncConfiguration   *config_ptr = &enc_handle_ptr->scs_instance_array[0]->scs_ptr->static_config;
+    //EbSvtAv1EncConfiguration   *config_ptr = &enc_handle_ptr->scs_instance_array[0]->scs_ptr->static_config;
 
-    eb_set_thread_management_parameters(config_ptr);
+    //eb_set_thread_management_parameters(config_ptr);
```
Comment (mafonsoN) [inline Source/Lib/Encoder/Globals/EbEncHandle.c]: Do you want to keep these lines commented out?

**Comment by hassount** [comment]
this is for the cs2 branch, not needed anymore

**Pull request closed without merging**